crmsh/corosync.py

```diff
@@ -811,20 +811,48 @@ class IPAlreadyConfiguredError(Exception):
     pass
 
 
-def add_node_ucast(IParray, node_id=None):
-
-    f = open(conf()).read()
-    p = Parser(f)
-
-    # to check if the same IP already configured
+def find_configured_ip(IParray):
+    """
+    find if the same IP already configured
+    If so, raise IPAlreadyConfiguredError
+    """
+    with open(conf()) as f:
+        p = Parser(f.read())
+
+    # get exist ip list from corosync.conf
     exist_iplist = []
-    for path in p.all_paths():
+    for path in set(p.all_paths()):
         if re.search('nodelist.node.ring[0-9]*_addr', path):
             exist_iplist.extend(p.get_all(path))
+
+    # all_possible_ip is a ip list to check whether one of them already configured
+    all_possible_ip = []
+    # get local ip list
+    local_ip_list = utils.ip_in_local(utils.is_ipv6(IParray[0]))
+    # extend all_possible_ip if IParray contain local ip
+    # to avoid this scenarios in join node:
+    #   eth0's ip already configured in corosync.conf
+    #   eth1's ip also want to add in nodelist
+    # if this scenarios happened, raise IPAlreadyConfiguredError
     for ip in IParray:
+        if ip in local_ip_list:
+            all_possible_ip = IParray + local_ip_list
+            break
+    else:
+        all_possible_ip = IParray
+
+    for ip in set(all_possible_ip):
```

Comment:
I think you can change this for loops by (if both are `sets`). This checks if some value is in both lists:
```
if bool(all_possible_ip & exist_iplist):
    raise IPAlreadyConfiguredError("IP {} was already configured".format(ip))
```